modules/hostap/src/supp_api.c

```diff
@@ -145,6 +145,53 @@ static void supp_shell_connect_status(struct k_work *work)
 	k_mutex_unlock(&wpa_supplicant_mutex);
 }
 
+static struct hostapd_hw_modes *get_mode_by_band(struct wpa_supplicant *wpa_s, uint8_t band)
+{
+	enum hostapd_hw_mode hw_mode;
+	bool is_6ghz = (band & BIT(WIFI_FREQ_BAND_6_GHZ)) ? true:false;
+
+	if ((band & BIT(WIFI_FREQ_BAND_2_4_GHZ))) {
```

Comment:
Just realized that the Wi-Fi management bands are not bitmaps, just normal enum values.

Reply:
Yes. Used BIT macro here to differentiate the case where no band is assigned (band initialized to 0 in configuration API) and the case where band is set to 2.4GHz.

Reply:
But the input is not modified `params->band` is a normal value not BIT()?

Reply:
We are setting BIT() for param band in wifi_ext_mgmt module:

```
if (creds->header.flags & WIFI_CREDENTIALS_FLAG_2_4GHz) {
		params->band |= BIT(WIFI_FREQ_BAND_2_4_GHZ);
	}

	if (creds->header.flags & WIFI_CREDENTIALS_FLAG_5GHz) {
		params->band |= BIT(WIFI_FREQ_BAND_5_GHZ);
	}
```

Reply:
That is wrong, as per docs `params->band` is just an enum, so, any driver will try to process it as enum and might fail. Though I don't see this param used currently.

Reply:
Do you see any issue if I change the enum definition i.e., set 2,4GHz to 1?

Reply:
Nope, its fine. But that would mean an upstream change :)

Reply:
I would still suggest that we use the BIT() approach, since having 0 as a valid value for a field in structure seems problematic. Specially in this case, since, for a structure instantiated to 0, the default band becomes 2.4GHz-only. 
Also, this avoids the upstream change.

Reply:
Sure, in that case then please use an intermediate variable rather than modify `params->band`, in fact `params` should be treated as `const` (missed this in the original implementation).

Reply:
As discussed, avoided usage of `BIT()` for `params->band`. Instead setting band to `WIFI_FREQ_BAND_UNKNOWN` in case band is not configure.